Chessington.GameEngine/Pieces/Rook.cs

```diff
@@ -10,7 +10,21 @@ public Rook(Player player)
 
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
-            return Enumerable.Empty<Square>();
+            List<Square> availableMoves = new List<Square>();
+            Square here = board.FindPiece(this);
+            foreach (var index in Enumerable.Range(0,8))
+            {
+                if (here.Col != index)
+                {
+                    availableMoves.Add(new Square(here.Row, index));
+                }
+
+                if (here.Row != index)
+                {
+                    availableMoves.Add(new Square(index,here.Col));
+                }
+            }
+            return availableMoves;
```

Comment:
Seems like tests are not accounting for it, but you will have Current Position (even twice) in your available moves list

Reply:
I don't think it should do, due to the "if (here.Col != index)"

Reply:
ah, nice, I missed that one 👍